### ISSUE TITLE
✍️ Contribution period: <Nwuguru Chidiebere> 

### ISSUE THREAD

**Issue by Nwuguru-Chidiebere-Sullivan** [ersilia-os/ersilia#1013]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [ ] Install the Ersilia Model Hub and test the simplest model
- [ ] Install Docker if needed, and test another model
- [ ] Write a motivation statement to work at Ersilia
- [ ] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [ ] Select a model from the list suggested in GitBook
- [ ] Download and serve the model via the Ersilia Model Hub to ensure it works
- [ ] Open a repository on your GitHub user with all the necessary files
- [ ] Select and clean a dataset of 1000 molecules (example notebook 1)
- [ ] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [ ] Find a suitable dataset with sufficient experimental results
- [ ] Clean and standardize the dataset
- [ ] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
![Screenshot 2024-03-08 041656](https://github.com/ersilia-os/ersilia/assets/129173970/5f61e904-39ad-4409-a99e-c5e98f3c0ed6)

Installing docker and connecting it to my terminal gave me a whole lot of difficulties; I spent two nights on it. It was this difficult because I was initially trying to run systemctl command in WSL environment. How I resolved this and was able to run docker from my Ubuntu terminal was:
- Install docker desktop
- Enable virtualisation on my computer
- Switched from WSL-1 to WSL-2 from my terminal and set the version to Ubuntu
- Enabled Ubuntu in my docker desktop settings

This way, I was able to run 'hello-world' with docker from my Ubuntu terminal and it went successfully.

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
![Screenshot 2024-03-08 044701](https://github.com/ersilia-os/ersilia/assets/129173970/18b64d21-834d-4816-b379-c1e6eed655cf)

Attached is the proof to show that Ersilia was recognised by my  CLI and was able to run a simple model as instructed.

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
<img width="371" alt="fetching from dockerhub" src="https://github.com/ersilia-os/ersilia/assets/129173970/8fd14f04-2be8-4bd7-b654-08d8f795fc05">

Attached is the proof showing that I was able to pull a model image,  test the mode I fetched from DockerHub, and was able to generate the Morgan Fingerprints for a molecule.

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
### Motivation Statement

It was in my last Molecular Biology class as a final-year medical student, as we were been taught Bioinformatics by a lecturer who I didn't quite like, especially because students barely make good grades in his courses, that I first heard about data analysis from this lecturer. I still remember his exact words as he asked, "Who in this class knows about  data analysis and how it applies to healthcare?" None of us in the class knew what it was, and it wasn't even a surprise because, in the part of Nigeria where I lived and schooled, the presence of tech industries can only be imagined, even at this point. After the class, I went home determined to research data analysis, especially since the lecturer did say a few things about it and how it applies to healthcare. During my research, I came across data science and loved it and how it applies to healthcare. Ever since then, the trajectory of my career as a young medical professional changed; I no longer wanted to practice directly from the hospital, I wanted to explore data science and proffer a more robust solution to many healthcare challenges by leveraging the power of actionable insights that can be uncovered from clinical data. This quest led me to a rigorous self-study that enabled me to acquire data analytical and machine learning skills.

Although I have been able to learn and acquire analytical skills such as Python, SQL, Excel, Tableau, ML, pandas, seaborn, matplotlib, scikit-learn, Git, MongoDB, Jupyter Notebook, Google, Colab, etc, the journey has never been easy and I still lack substantial work experience, especially for someone who comes from a technologically disadvantaged area; I needed an enabling environment that would allow me to implement my skills, collaborate with others, and further learn from more experienced data professionals. So late last year, I came across Outreachy internship and how it supports technologically disadvantaged individuals in gaining substantial experience through their internship; getting into Outreachy internship became my big dream because that was the way I could get to finally advance my tech journey, especially as someone without a traditional tech background.

On getting into Outreachy contribution phase, I came across Ersilia and the project resonated so much with me. As someone whose long-term is to gather enough experience and become a healthcare data scientist to reckon with, a project geared towards ready-to-use AI/ML models for biomedical research was just everything I needed at this stage. Working with Ersilia will not only help advance my career but will inculcate a culture of collaboration at a high level and provide the enabling environment where I can put into practice the skills I have learned so far, thus, helping to shape my future as a healthcare data scientist, while in return, I will play a part in advancing the course of Ersilia project through my meaningful contributions, especially as it concerns improving healthcare system through Artificial Intelligence and Machine Learning. I am confident that with my medical background and technical skills, I will make a meaningful contribution to this project.

**Comment by DhanshreeA** [comment]
@Nwuguru-Chidiebere-Sullivan Quick question, if you have access to an Ubuntu machine why not work on that? WSL has been more painful to work with in our experience as well.

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
@DhanshreeA Yeah, I am currently working directly from an Ubuntu machine. I just don't know for what reason my computer was finding it difficult to integrate docker into my docker terminal until I had done it from WSL, but now it's working well in my Ubuntu terminal.

**Comment by GemmaTuron** [comment]
Hi @Nwuguru-Chidiebere-Sullivan 

You have ticked off W2 tasks but we have not seen updates here. Please can you detail the tasks and link to your repository so we can provide feedback?

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
Hi @GemmaTuron, I had a hard time concluding the tasks for Week 2 but I'm glad I persisted, it allowed me to learn a lot through the difficulties. 

Here is the link to my [GitHub Repo](https://github.com/Nwuguru-Chidiebere-Sullivan/Outreachy-Ersilia-Project-Week2-Tasks) containing my detailed works on the week 2 tasks. I shall also detail my encounters, struggles, and findings from the week 2 tasks in the section below. Please, I look forward to your feedback so I can be able to move on to week 3 tasks, especially since time is almost running against me. Thank you!

**Comment by GemmaTuron** [comment]
Hi @Nwuguru-Chidiebere-Sullivan,

Good work, I don't have any specific feedback, just a clarification. task 3 actually refers to week 3 task where you are validating the model externally? Please clarify this and  then move onto preparing your final application

**Comment by Nwuguru-Chidiebere-Sullivan** [comment]
> Hi @Nwuguru-Chidiebere-Sullivan,
> 
> Good work, I don't have any specific feedback, just a clarification. task 3 actually refers to week 3 task where you are validating the model externally? Please clarify this and then move on to preparing your final application

Thank you so much @GemmaTuron and @DhanshreeA for the feedback. I have indicated the task for week 3 as you instructed in my notebook and ReadMe.
